addons/necropost-highlighter/userscript.js

```diff
@@ -0,0 +1,333 @@
+    if (!title) {
+      return "";
+    }
+    let titleText = title.innerText;
+    if (titleText.includes(searchResultsPageName)) {
+      return searchResultsPageName;
+    }
+    // check each forum we explicitly know about, so might be highlighting
+    for (let setting in customForumSettingToForum) {
+      let forum = customForumSettingToForum[setting];
+      if (titleText.includes(forum)) {
+        return forum;
+      }
+    }
+    // a page we don't know how to highlight (and probably shouldn't have run on)
+    return "";
+  }
+
+  /**
+   * Search Results or user selected forums, or default forums if custom forums is off
+   */
+  async function isOnSelectedForumPage() {
+    if (forumOrSearchPageName.includes(searchResultsPageName)) {
+      return true;
+    }
+    let activatedForums = determineActivatedForums();
+    for (const activatedForum of activatedForums) {
+      if (forumOrSearchPageName.includes(activatedForum)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Does not conflict with Addon: Website dark mode and customizable colors
+   */
+  function highlightTopicCellsWithTopicIdBelow(topics, lowestNewTopicId) {
+    const standardSiteStickyClass = ".isticky";
+    const mobileSiteStickyClassName = "sticky";
+
+    let activatedForums = determineActivatedForums();
+    topics
+      .filter((topic) => !topic.topicCell.querySelector(standardSiteStickyClass)) // on a child
+      .filter((topic) => !topic.topicCell.classList.contains(mobileSiteStickyClassName)) // on this item
+      .filter((topic) => topic.topicId < lowestNewTopicId)
+      .filter((topic) => activatedForums.includes(topic.forum))
+      .forEach(highlightSingle);
+
+    function highlightSingle(topic) {
+      if (!topic.restoreCell) {
+        topic.restoreCell = topic.topicCell.cloneNode(true);
+      }
+      const necropostMessage = msg("(Necropost?)");
+      const highlightColor = addon.settings.get("highlightColor");
+
+      topic.topicCell.style.backgroundColor = highlightColor;
+      if (isMobileSite) {
+        let replies = topic.topicCell.querySelector("span");
+        replies.textContent += " " + necropostMessage;
+        return;
+      }
+      let possibleNewPostsLink = topic.topicCell.querySelector(".tclcon>a");
+      if (!possibleNewPostsLink) {
+        // No New Posts link. Insert a plain text '(Necropost?)'
+        let byUser = topic.topicCell.querySelector("span.byuser");
+        byUser.textContent += " " + necropostMessage;
+      } else {
+        possibleNewPostsLink.textContent = necropostMessage;
+      }
+    }
+  }
+
+  function highestTopicIdFrom(topics) {
+    let highestSoFar = -1;
+    for (let i = 0; i < topics.length; i++) {
+      if (topics[i].topicId > highestSoFar) {
+        highestSoFar = topics[i].topicId;
+      }
+    }
+    return highestSoFar;
+  }
+
+  /**
+   * Returns int topicId, or 0 if no topic was found.
+   */
+  async function extractTopicIdFrom(topicCell) {
```

Comment:
Most of the functions in this addon really do not need async.
```suggestion
  function extractTopicIdFrom(topicCell) {
```